.github/workflows/pull-request.yaml

```diff
@@ -25,78 +25,87 @@ jobs:
-          cd ${{ env.FLYWAY_PATH }}
-          curl -L https://repo1.maven.org/maven2/org/flywaydb/flyway-commandline/${FLYWAY_VERSION}/flyway-commandline-${FLYWAY_VERSION}.tar.gz -o out.tar.gz
-          tar --strip-components=1 -xzf out.tar.gz
-          rm out.tar.gz
-          ls -al
-      - name: Cache Hardhat
-        id: cache-hardhat
-        uses: actions/cache@v2
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt
+      # Start the build process in the background. The following cargo test command will automatically
+      # wait for the build process to be done before proceeding.
+      - run: cargo build -p orderbook -p database --tests &
+      - uses: yu-ichiro/spin-up-docker-compose-action@v1
         with:
-          path: ${{ env.HARDHAT_PATH_ }}/node_modules
-          key: ${{ runner.os }}-hardhat-${{ env.HARDHAT_VERSION_ }}
-      - name: Install Hardhat
-        if: steps.cache-hardhat.outputs.cache-hit != 'true'
-        run: |
-          mkdir --parents ${{ env.HARDHAT_PATH_ }}
-          cd ${{ env.HARDHAT_PATH_ }}
-          npm install hardhat@${{ env.HARDHAT_VERSION_ }}
-          ls -al
-      - name: Setup Hardhat
-        working-directory: ${{ env.HARDHAT_PATH_ }}
-        run: |
-          cp ${{ github.workspace }}/docker/hardhat/hardhat.config.js hardhat.config.js
-          node_modules/.bin/hardhat node --hostname 127.0.0.1 &
-      - name: Setup Database
-        run: |
-          sudo systemctl start postgresql.service
-          sudo -u postgres createuser $USER
-          sudo -u postgres createdb $USER
-          psql -c "ALTER USER $USER PASSWORD '$FLYWAY_PASSWORD';"
-          ${{ env.FLYWAY_PATH }}/flyway -url="jdbc:postgresql:///" -user=$USER -locations="filesystem:database/sql/" migrate
-      - run: cargo test --no-run
-      - run: cargo test
-      # tests requiring a Postgres instance.
-      - run: cargo test postgres -- --ignored --test-threads 1
-      # tests requiring a local ethereum node
-      # nocapture because tracing output isn't captured by the test runner which makes it confusing
-      # to debug failing e2e tests.
-      - run: cargo test local_node -- --ignored --test-threads 1 --nocapture
-  openapi:
+          file: docker-compose.yaml
+          up-opts: -d db migrations
+      - run: cargo test postgres -p orderbook -p database -- --ignored --test-threads 1
+
+  test-local-node:
     timeout-minutes: 60
     runs-on: ubuntu-latest
+    env:
+      # Shrink artifact size by not including debug info. Makes build faster and shrinks cache.
+      CARGO_PROFILE_DEV_DEBUG: 0
+      CARGO_PROFILE_TEST_DEBUG: 0
     steps:
       - uses: actions/checkout@v2
-      - run: npm install @apidevtools/swagger-cli
-      - run: node_modules/.bin/swagger-cli validate crates/orderbook/openapi.yml
-      - run: node_modules/.bin/swagger-cli validate crates/driver/openapi.yml
-      - run: node_modules/.bin/swagger-cli validate crates/solvers/openapi.yml
```

Comment:
I think we should probably keep the openapi test

Reply:
I removed this accidentally :smile:

Reply:
https://github.com/cowprotocol/services/pull/1503/commits/22fa247c1df91a99a1deeda3b38c53ba73866d7d